analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -253,12 +253,12 @@ open class Npm(
                     OkHttpClientHelper.downloadText("$npmRegistry/$encodedName/$version").onSuccess {
                         val versionInfo = jsonMapper.readTree(it)
 
-                        description = versionInfo["description"].textValueOrEmpty()
-                        homepageUrl = versionInfo["homepage"].textValueOrEmpty()
+                        versionInfo["description"]?.textValue()?.takeUnless { it.isEmpty() }?.also { description = it }
+                        versionInfo["homepage"]?.textValue()?.takeUnless { it.isEmpty() }?.also { homepageUrl = it }
 
                         versionInfo["dist"]?.let { dist ->
-                            downloadUrl = dist["tarball"].textValueOrEmpty()
-                            hash = Hash.create(dist["shasum"].textValueOrEmpty())
+                            dist["tarball"]?.textValue()?.takeUnless { it.isEmpty() }?.also { downloadUrl = it }
+                            dist["shasum"]?.textValue()?.takeUnless { it.isEmpty() }?.also { hash = Hash.create(it) }
```

Comment:
The values should always be used together, in theory it is now possible that only one of them is updated, resulting in the download URL and hash not matching.